### PR TITLE
Clean deprecated context.StopCh

### PR DESCRIPTION
**Pull request by mjudeikis**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Related to old PR: https://github.com/kubernetes/kubernetes/pull/122148
Cleaning deprecated channel inside startup hooks in favour of ctx itself.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
clean deprecated context.StopCh in favor of ctx
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go [hunk 1]
```diff
@@ -300,7 +300,7 @@ func (c completedConfig) NewWithDelegate(delegationTarget genericapiserver.Deleg
 			ctx, cancel := context.WithCancel(context.Background())
 			go func() {
 				select {
-				case <-postStartHookContext.StopCh:
+				case <-postStartHookContext.Done():
```
Comment (mjudeikis) [inline staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go:303]: @sttts  do you know what is happening here? L298-L308. We basically constructed a new background ctx and used it, over post start one. This does not look right, but not sure I want to change it now.

Comment (sttts) [reply staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go:303]: good question. I don't understand the comment either.

**Comment by sttts** [comment]
Sgtm.

File: staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go (hunk 1 above)
Comment (sttts) [reply staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go:303]: actually, the comment is the same in the other cases. Here, the helper is not used. Maybe just an oversight at the time the helper was introduced. Worth to check the PR when it was added.

Comment (sttts) [reply staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go:303]: So, I think it is safe to change this one too.

Comment (pohly) [reply staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go:303]: This looks like leftover code: `postStartHookContext` used to have a stop channel. This function here then constructed a context from that (it should have done that with `wait.ContextForChannel`, but that's a different story).

Now `postStartHookContext` itself is a proper context.

So removing `ctx` and replacing it with `postStartHookContext` should be okay.

Comment (mjudeikis) [reply staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go:303]: ack. Will make the change too. It looked strange. So was thinking if we have cases where ctx gets terminated and we need it not to be or smth similar.

**Comment by sttts** [comment]
Needs rebase.

/lgtm
/approve

**Comment by sttts** [comment]
@liggitt ptal. pkg/registry changes.

File: pkg/controlplane/apiserver/server.go [hunk 2]
```diff
@@ -139,7 +138,7 @@ func (c completedConfig) New(name string, delegationTarget genericapiserver.Dele
 	}
 	if len(c.SystemNamespaces) > 0 {
 		s.GenericAPIServer.AddPostStartHookOrDie("start-system-namespaces-controller", func(hookContext genericapiserver.PostStartHookContext) error {
-			go systemnamespaces.NewController(c.SystemNamespaces, client, s.VersionedInformers.Core().V1().Namespaces()).Run(hookContext.StopCh)
+			go systemnamespaces.NewController(c.SystemNamespaces, client, s.VersionedInformers.Core().V1().Namespaces()).Run(hookContext.Done())
```
Comment (pohly) [inline pkg/controlplane/apiserver/server.go:141]: Thinking ahead: for [contextual logging](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/3077-contextual-logging) what we really want to pass around is the `context.Context` itself, not it's `Done` channel.

This cannot be done (sic) without API changes, so it's more work and leads to a bigger PR.

It then will need to touch the same lines again. Would it make sense to extend the scope of this PR? I'm fine either way, I just wanted to point out that this is not necessarily the final state.

Comment (mjudeikis) [reply pkg/controlplane/apiserver/server.go:141]: But this means we need to change each `Run` function for this? 
I would make them separate PRs if we can. Smaller is always better :D

Comment (pohly) [reply pkg/controlplane/apiserver/server.go:141]: For private code, yes, `Run` would need to be changed.

For public code it's worse: most likely we will need a `RunWithContext` alternative.

**Pull request merged**